src/meta.ml

```diff
@@ -207,24 +212,29 @@ let builtins ~stdlib_dir =
     }
   in
   let compiler_libs =
-    let sub name deps =
-      Package (simple name deps ~archive_name:("ocaml" ^ name))
+    let sub ?dir name deps =
+      Package (simple name ?dir deps ~archive_name:("ocaml" ^ name))
     in
     { name = "compiler-libs"
     ; entries =
         [ requires []
         ; version
-        ; directory "+compiler-libs"
-        ; sub "common" []
-        ; sub "bytecomp" ["compiler-libs.common"  ]
-        ; sub "optcomp"  ["compiler-libs.common"  ]
+        ]
+        @ (if sep_libraries then [] else [ directory "+compiler-libs" ])
+        @
+        [ sub "common" []
```

Comment:
ocamlcommon is not in a sub-directory as well?

Reply:
It is, I guessed I missed this change and for some reason, it didn't break anything.

Reply:
Fixed now.